build_node/utils/codenotary.py

```diff
@@ -0,0 +1,73 @@
+import typing
+
+from cas_wrapper import CasWrapper, CasArtifact
+import rpm
+
+from build_node.models import Task, Artifact
+
+
+__all__ = [
+    'notarize_build_artifacts',
+]
+
+
+def notarize_build_artifacts(
+    task: Task,
+    build_artifacts: typing.List[Artifact],
+    cas_client: CasWrapper,
+    build_host: str,
+):
+    cas_metadata = {
+        'build_id': task.build_id,
+        'build_host': build_host,
+        'build_arch': task.arch,
+        'built_by': task.created_by.full_name,
+        'sbom_api': '0.1',
+    }
+    if task.is_alma_source() and task.alma_commit_cas_hash:
+        cas_metadata['alma_commit_sbom_hash'] = task.alma_commit_cas_hash
+    if task.ref.git_ref:
+        cas_metadata.update({
+            'source_type': 'git',
+            'git_url': task.ref.url,
+            'git_ref': task.ref.git_ref,
+            'git_commit': task.ref.git_commit_hash,
+        })
+    else:
+        srpm = next((
+            artifact for artifact in build_artifacts
+            if artifact.name.endswith('.src.rpm')
+        ), None)
+        if srpm:
+            ts = rpm.TransactionSet()
+            with open(srpm.path, 'rb') as rpm_pkg:
+                hdr = ts.hdrFromFdno(rpm_pkg)
+                srpm_nevra = (
+                    f"{hdr['epoch']}:{hdr['name']}-{hdr['version']}-"
+                    f"{hdr['release']}.src"
+                )
```

Comment:
This is interesting, we can benefit from it in future if you will mote it to function like `"get_rpm_metadata()"`